projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/components/unlink-cases/unlink-cases.component.ts

```diff
@@ -0,0 +1,146 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { FormArray, FormBuilder, FormGroup } from '@angular/forms';
+import { forkJoin } from 'rxjs';
+import { CaseView, ErrorMessage } from '../../../../../domain';
+import { CasesService } from '../../../../case-editor/services/cases.service';
+import { CaseLink, LinkedCasesState } from '../../domain';
+import { LinkedCasesErrorMessages, LinkedCasesPages } from '../../enums/write-linked-cases-field.enum';
+import { LinkedCasesService } from '../../services/linked-cases.service';
+
+@Component({
+  selector: 'ccd-unlink-cases',
+  templateUrl: './unlink-cases.component.html'
+})
+export class UnLinkCasesComponent implements OnInit {
+
+  private static readonly LINKED_CASES_TAB_ID = 'linked_cases_sscs';
+  private static readonly CASE_NAME_MISSING_TEXT = 'Case name missing';
+
+  @Output()
+  public linkedCasesStateEmitter: EventEmitter<LinkedCasesState> = new EventEmitter<LinkedCasesState>();
+  @Output()
+  public notifyAPIFailure: EventEmitter<boolean> = new EventEmitter(false);
+
+  public unlinkCaseForm: FormGroup;
+  public caseId: string;
+  public linkedCases: CaseLink[] = [];
+  public errorMessages: ErrorMessage[] = [];
+  public unlinkErrorMessage: string;
+  public isLoaded: boolean;
+  public isServerError = false;
+
+  constructor(private readonly fb: FormBuilder,
+    private readonly casesService: CasesService,
+    private readonly linkedCasesService: LinkedCasesService) {
+  }
+
+  public ngOnInit(): void {
+    this.getLinkedCases();
+  }
+
+  public getLinkedCases(): void {
+    this.caseId = this.linkedCasesService.caseId;
+    if (this.linkedCasesService.linkedCases.length > 0) {
+      this.linkedCases = this.linkedCasesService.linkedCases;
+      this.getAllLinkedCaseInformation();
+    } else {
+      this.casesService.getCaseViewV2(this.caseId).subscribe((caseView: CaseView) => {
+        const linkedCasesTab = caseView.tabs.find(tab => tab.id === UnLinkCasesComponent.LINKED_CASES_TAB_ID);
+        if (linkedCasesTab) {
+          const linkedCases: CaseLink[] = linkedCasesTab.fields[0].value;
+          this.linkedCases = linkedCases;
+          this.linkedCasesService.linkedCases = linkedCases;
+          this.getAllLinkedCaseInformation();
+        }
+      });
+    }
+  }
+
+  public getAllLinkedCaseInformation() {
```

Comment:
@johnbenjamin-hmcts - please specify return type if possible

Reply:
@Daniel-Lam I have updated the code now.